File: app/src/main/java/com/example/m20210712_mubasharkhan_nycschools/repository/Repository.java
package com.example.m20210712_mubasharkhan_nycschools.repository;

import com.example.m20210712_mubasharkhan_nycschools.model.School;
import com.example.m20210712_mubasharkhan_nycschools.network.ApiResponse;

import java.util.List;

public interface Repository {
    ApiResponse<List<School>> loadAllSchools();
}

File: app/src/main/java/com/example/m20210712_mubasharkhan_nycschools/repository/Repository.kt
package com.example.m20210712_mubasharkhan_nycschools.repository

import com.example.m20210712_mubasharkhan_nycschools.model.SatScore
import com.example.m20210712_mubasharkhan_nycschools.model.School

interface Repository {
   suspend fun loadAllSchools(): List<School>
   suspend fun loadSatScore(dbn: String): List<SatScore>
}